BLSQ/iaso
# POLIO-1449 vaccine stock management backend form a destruction incident api

Adding endpoints to create/list/edit OutgoingStockMovements (FormA), Destruction Reports and Incident Reports.

⚠️ Please ignore the frontend part of this PR. It's in another PR specifically for the front and is gonna be reviewed independantly, see here : #1098

Related JIRA tickets : POLIO-1449

## Self proofreading checklist


## Changes

- Create the viewsets and serializers for `OutgoingStockMovement`
- Create the viewsets and serializers for `IncidentReport`
- Create the viewsets and serializers for `DestructionReport`.
- Refactoring of the stock calculations out of VaccineStock serializer and into a custom class `VaccineStockCalculator` thats used in the serializer and in the VaccineStock Viewset.
- Add inline documentations and comments.
- Added more tests.
- Migration to allow multiple "lot_numbers" on DestructionReport

## How to test

See tests in `plugins/polio/tests/test_vaccine_stock_management.py`
You need to have all the correct objects created locally in order to be able to test via swagger or the frontend. See the description in #1081 for details on what to create.
